samples/Mvc.Client/Controllers/AuthenticationController.cs

```diff
@@ -18,8 +18,11 @@ public class AuthenticationController : Controller
         public async Task<IActionResult> SignIn() => View("SignIn", await HttpContext.GetExternalProvidersAsync());
 
         [HttpPost("~/signin")]
-        public async Task<IActionResult> SignIn([FromForm] string provider)
+#pragma warning disable CA1054 // Uri parameters should not be strings
+        public async Task<IActionResult> SignIn([FromForm] string provider, [FromForm] string? returnUrl)
+#pragma warning restore CA1054 // Uri parameters should not be strings
         {
+            returnUrl ??= "/";
```

Comment:
This appears to allow an open redirect. It should only allow the return URL to be a local path on the same domain.